e2e/testcases/ssh_known_hosts_test.go

```diff
@@ -0,0 +1,204 @@
+		},
+	)
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+
+	// try syncing resource and validate
+	nt.Must(nt.RootRepos[configsync.RootSyncName].Copy(fmt.Sprintf("%s/accounting-namespace.yaml", yamlDir), "acme/namespaces/accounting/namespace.yaml"))
+	nt.Must(nt.RootRepos[configsync.RootSyncName].CommitAndPush("Add accounting namespace"))
+	if err := nt.WatchForAllSyncs(); err != nil {
+		nt.T.Fatal(err)
+	}
+	if err := nt.Validate("accounting", "", &corev1.Namespace{}); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	nt.Must(nt.RootRepos[configsync.RootSyncName].Remove("acme/namespaces/accounting/namespace.yaml"))
+	nt.Must(nt.RootRepos[configsync.RootSyncName].CommitAndPush("Remove accounting namespace"))
+	if err := nt.WatchForAllSyncs(); err != nil {
+		nt.T.Fatal(err)
+	}
+	if err := nt.ValidateNotFound("accounting", "", &corev1.Namespace{}); err != nil {
+		nt.T.Fatalf("error: %s", err)
+	}
+
+	// validate root sync error using invalid known host value
+	knownHostValue = "invalid value"
+	nt.MustMergePatch(rootSecret, secretDataPatch(controllers.KnownHostsKey, knownHostValue))
+	err = nt.Watcher.WatchObject(kinds.Deployment(),
+		core.RootReconcilerPrefix, configsync.ControllerNamespace,
+		[]testpredicates.Predicate{
+			testpredicates.DeploymentHasEnvVar(reconcilermanager.GitSync, controllers.GitSyncKnownHosts, "true"),
+		},
+	)
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+	err = nt.Watcher.WatchObject(kinds.RootSyncV1Beta1(), configsync.RootSyncName, configsync.ControllerNamespace, []testpredicates.Predicate{
+		testpredicates.RootSyncHasSourceError(status.SourceErrorCode, "No ED25519 host key is known"),
+	})
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+}
+
+func TestRepoReconcilerSSHKnownHost(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.SyncSource, ntopts.SkipNonLocalGitProvider,
+		ntopts.NamespaceRepo(backendNamespace, configsync.RepoSyncName), ntopts.WithDelegatedControl,
+		ntopts.RepoSyncPermissions(policy.AppsAdmin(), policy.CoreAdmin()))
+	var err error
+	repoSecret := &corev1.Secret{}
+	repoReconcilerName := core.NsReconcilerName(backendNamespace, configsync.RepoSyncName)
+	repoSyncNN := nomostest.RepoSyncNN(backendNamespace, configsync.RepoSyncName)
+	nt.T.Cleanup(func() {
+		nt.MustMergePatch(repoSecret, secretDataDeletePatch(controllers.KnownHostsKey))
+		nomostest.DeletePodByLabel(nt, "app", reconcilermanager.ManagerName, true)
+	})
+
+	// validate secret and deployment initial stage
+	err = nt.Validate(nomostest.NamespaceAuthSecretName, backendNamespace, repoSecret)
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+	err = nt.Watcher.WatchObject(kinds.Deployment(),
+		repoReconcilerName, configsync.ControllerNamespace,
+		[]testpredicates.Predicate{
+			testpredicates.DeploymentHasEnvVar(reconcilermanager.GitSync, controllers.GitSyncKnownHosts, "false"),
+		},
+	)
+	if err != nil {
+		nt.T.Fatal(err)
+	}
+
+	// validate repo sync error using invalid known host value
```

Comment:
Suggest ordering the steps the same way for both tests, so that it is easier to follow when reading each test case

Reply:
both tests have been streamlined to follow the same steps and ordering